### ISSUE TITLE
improving location name (beijing, mumbai etc) recognition accuracy

### ISSUE THREAD

**Issue by cogmeta** [PaddlePaddle/DeepSpeech#150]
The default model is horrible at recognising location names. Any idea how that can be improved? I tried with custom language models but the accuracy is not good. Any suggestions

**Comment by kuke** [comment]
@cogmeta The Mandarin model lacks of large training data for acoustic model and language model. If we can work out that in the future, the model maybe will have a significant improvement in accuracy including recognizing location name.

**Comment by cogmeta** [comment]
Thank you. Is is possible to build a custom decoder that can take a list of words (not necessarily seen in training set) and spit out the most probable one? That way, we can input a list of city names to the decode and the find the most probable one?

**Comment by cogmeta** [comment]
When you use the "ctc_greedy" method the output is very much similar to the city names so i assume the custom decoder approach should work. Right?

**Comment by kuke** [comment]
@cogmeta If it is an isolated word recognition system, your approach should be able to work. But for the continuous speech recognition, I am not sure. A bad case is that for some words just having similar pronunciation with cities' name, they will be easily recognized mistakenly.

**Comment by cogmeta** [comment]
Yes. For now, I am trying to get isolated world recognition. That should be get me started at least. I am totally clueless as of now how that can be achieved into continuous speech recognition. Any ideas? Thanks.

**Comment by cogmeta** [comment]
Is it possible to break the voice input into separate chunks (using the silent periods of the voice sample as breakpoints) and then do the speech recognition? Can you recommend a good algorithm or library to create the separate chunks?

**Comment by kuke** [comment]
In principle yes, but in reality it is very difficult to collect massive clean training data in this way.  We are not expert in isolated word recognition. Maybe you can turn to some research paper for help.

**Comment by cogmeta** [comment]
Got it, Kuke. Thanks for the help. We will update if you end up getting some good results in that direction.
